codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -296,6 +296,7 @@ final DefaultPropertyKey verifyKey(PropertyKey key) {
         private int totalChildWeights;
         private int prioritizableForTree = 1;
         private boolean resetSent;
+        private boolean headerSent;
```

Comment:
@nmittler - The flow controller would still be told what the frame types are via `FlowControlled`. The implementation for `isInitialHeadersQueued` may look something like this:

```java
public boolean isInitialHeadersQueued(Stream stream) {
  State state = state(stream);
  if (state == null) {
    return false;
  }
  FlowControlled firstFrame = state.pendingWriteQueue.peekFirst(); /* we could add a method to AbstractState to do this in case pendingWriteQueue doesn't exist */
  return firstFrame != null && firstFrame.isInitialHeaders();
}
```

If the `Headers` portion of the name is bringing too much frame specific knowledge into the flow controller we could rename it to something line `isInitialFrame()` and `isInitialFrameQueued()`.